includes/class-wc-gateway-amazon-payments-advanced-abstract.php

```diff
@@ -969,4 +971,50 @@ public function has_other_gateways_enabled() {
 		}
 		return false;
 	}
+
+	/**
+	 * Displays an explanation/warning in order to make it clear how Amazon Pay
+	 * treats postcode restrictions.
+	 *
+	 * @param WC_Shipping_Zone $zone
+	 * @return void
+	 */
+	public function postcode_explanation_message( $zone ) {
+		$pcodes = false;
+
+		foreach ( $zone->get_zone_locations() as $location ) {
+			if ( 'postcode' === $location->type ) {
+				$pcodes = true;
+				break;
+			}
+		}
+
+		if ( $pcodes ) :
+			?>
+			<div class="notice notice-warning">
+				<p>
+					<strong><?php esc_html_e( 'Amazon Pay Notice:', 'woocommerce-gateway-amazon-payments-advanced' ); ?></strong> 
+					<?php
+					/* translators: 1) Opening b tag. 2) Closing b tag. */
+					printf( esc_html__( 'When using postcodes to restrict shipping locations, a selected zone is required for them to apply and they will apply to %1$sall%2$s the selected zones. So please, restrict accordingly.', 'woocommerce-gateway-amazon-payments-advanced' ), '<b>', '</b>' );
+					?>
+				</p>
+				<p>
+					<?php
+					/* translators: 1) Opening b tag. 2) Closing b tag. */
+					printf( esc_html__( 'Additionally, be careful when specifying ranges. By default WooCommerce supports only %1$sfully numeric ranges%2$s. The same applies for Amazon Pay.', 'woocommerce-gateway-amazon-payments-advanced' ), '<b>', '</b>' );
+					?>
+				</p>
+				<p>
+					<?php
+					esc_html_e(
+						'Ranges covering thousands or more of postcodes may slow down your site, since Amazon Pay does not support postcode ranges we loop from the minimum until we reach the maximum in order to include them all. Try to use the asterisk (*) wildcard when possible for faster loading times.',
```

Comment:
We should test with a large number of postcodes (99k ?) to make sure it works.
If it ends up clogging the payload & making API requests fail, it may be better to try something different.

Reply:
You are correct, when using 99k zip codes as range Amazon API fails. Will try to think of an alternative.

Reply:
Hmm, that's potentially a blocker. 

One option would be to notify the user that their settings are "incompatible" with what Amazon supports, and not include those postcodes in the API call - so that it doesn't fail. 
But again we would need to know/find what the threshold should be at which we are triggering that scenario.

Reply:
Implemented an algorithm where ranges are converted to wildcards using the '?' symbol supported by the Amazon API. Let me know your thoughts